sql/_05_plcsql/_01_testspec/_02_declaration/_01_parameter/_03_type_conversion/cases/02_01_03-13_sql_to_param_CHAR.sql

```diff
@@ -0,0 +1,208 @@
+end;
+call t_SHORT_CHAR('SHORT', 'CHAR', cast( 8934 as short ) ) ;
+drop procedure t_SHORT_CHAR ;
+
+
+call print_message('t_BIT(8)_CHAR. This scenario is a failure.');
+create or replace procedure t_BIT_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_BIT_CHAR('BIT(8)', 'CHAR', 0xaa ) ;
+drop procedure t_BIT_CHAR ;
+
+
+call print_message('t_BIT VARYING_CHAR. This scenario is a failure.');
+create or replace procedure t_BITVARYING_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_BITVARYING_CHAR('BIT VARYING', 'CHAR', 0xaa ) ;
+drop procedure t_BITVARYING_CHAR ;
+
+
+call print_message('t_CHAR_CHAR. This scenario is a success.');
+create or replace procedure t_CHAR_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_CHAR_CHAR('CHAR', 'CHAR', 'CHAR CUBRID' ) ;
+drop procedure t_CHAR_CHAR ;
+
+
+call print_message('t_VARCHAR_CHAR. This scenario is a success.');
+create or replace procedure t_VARCHAR_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_VARCHAR_CHAR('VARCHAR', 'CHAR', 'VARCHAR CUBRID' ) ;
+drop procedure t_VARCHAR_CHAR ;
+--BUG
+
+call print_message('t_SET_CHAR. This scenario is a failure.');
+create or replace procedure t_SET_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_SET_CHAR('SET', 'CHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_SET_CHAR ;
+--BUG
+
+call print_message('t_MULTISET_CHAR. This scenario is a failure.');
+create or replace procedure t_MULTISET_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_MULTISET_CHAR('MULTISET', 'CHAR', {'c','c','c','b','b','a'} ) ;
+drop procedure t_MULTISET_CHAR ;
+--BUG
+
+call print_message('t_LIST_CHAR. This scenario is a failure.');
+create or replace procedure t_LIST_CHAR(sql_type string, procedure_type string, param CHAR ) as begin
+    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param ); 
+end;
+call t_LIST_CHAR('LIST', 'CHAR', {'c','c','c','b','b', 'a'} ) ;
+drop procedure t_LIST_CHAR ;
+--BUG
+
+call print_message('t_ENUM_CHAR. This scenario is a failure.');
```

Comment:
It's a failure, but the answer file it was successful. Need to check.

Reply:
```suggestion
call print_message('t_ENUM_CHAR. This scenario is a success.');
```
Note it is a success as shown below
```
csql> select cast (cast('yellow' as enum ('red', 'yellow')) as char(10));

=== <Result of SELECT Command in Line 1> ===

   cast( cast('yellow' as enum('red', 'yellow')) as char(10))
======================
  'yellow    '        

```

Reply:
해당 케이스는 ENUM타입을 테스트 하는 것이 목적이므로 아래와 같이 변경하였습니다.
ENUM 컬럼타입의 테이블을 만들고, ENUM컬럼을 파라미터로 전달하여 확인하는 방법으로 변경하였습니다.

call print_message('t_ENUM_CHAR. This scenario is a failure.');
create or replace function t_ENUM_CHAR(sql_type string, procedure_type string, param CHAR ) return varchar(100) as begin
    dbms_output.put_line('sql_type = ' ||sql_type ||', procedure_type = '||procedure_type||', current_value = '|| param );
    return 'abc';
end;

CREATE TABLE enum_tbl (
    color ENUM ('red', 'yellow', 'blue', 'green')
);
INSERT INTO enum_tbl (color) VALUES ('yellow');

select t_ENUM_CHAR( 'ENUM', 'CHAR', color ) from enum_tbl;
drop function t_ENUM_CHAR ;
drop table enum_tbl;